worktree_status.go

```diff
@@ -409,6 +585,43 @@ func (w *Worktree) deleteFromFilesystem(path string) error {
 	return err
 }
 
+// RemoveGlob given a glob pattern removes all the matching files content and
+// all his sub-directories recursively in the worktree to the index.
+func (w *Worktree) RemoveGlob(pattern string) error {
+	files, err := util.Glob(w.Filesystem, pattern)
+	if err != nil {
+		return err
+	}
+
+	if len(files) == 0 {
+		return ErrGlobNoMatches
+	}
+
+	idx, err := w.r.Storer.Index()
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		fi, err := w.Filesystem.Lstat(file)
+		if err != nil {
+			return err
+		}
+
+		if fi.IsDir() {
+			_, err = w.doRemoveDirectory(idx, file)
+		} else {
+			_, err = w.doRemoveFile(idx, file)
+		}
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return w.r.Storer.SetIndex(idx)
+}
+
 // Move moves or rename a file in the worktree and the index, directories are
 // not supported.
 func (w *Worktree) Move(from, to string) (plumbing.Hash, error) {
```

Comment:
Are we planning to wait for now on adding support for directories to this method?

Reply:
Add TODO